### DISCUSSION TITLE
NumericUpDown custom unit

### DISCUSSION THREAD

**Discussion by dubrovenskaya** [AvaloniaUI/Avalonia#8723] [Q&A]
![NumericUpDownWithoutCustomUnit](https://user-images.githubusercontent.com/33326526/184106400-f4b181c8-c0f8-4b39-87ec-442833f951f6.png)
Hello together,
is there any possibility to bring a custom unit in the numeric up down control? 

P.S. The author is a complete junior. 

Thank you for any hints.

**Comment by timunie** [comment]
Iirc you can set a string format or InnerRightContent. But I cannot test it yet

**Comment by dubrovenskaya** [comment]
![NumericUpDown1](https://user-images.githubusercontent.com/33326526/184316206-cd4cd55e-8a6b-40f2-938c-b0bcd254aeda.png)
![NumericUpDown2](https://user-images.githubusercontent.com/33326526/184316219-325aded8-a7f5-43bf-90b8-c948b5eaaccb.png)
I have solved this as follows: on the two images, there are 2 different font themes to see.
Now I do not understand how to style the unit extra. When changing the property of the font, it is applied to the whole text box and not only to the unit. Are there any ideas how to solve this?

**Reply by timunie** [reply]
Make a new Style with the selector="TextBox.MyClassName" and you specific setter goes there

**Reply by dubrovenskaya** [reply]
> Make a new Style with the selector="TextBox.MyClassName" and you specific setter goes there

Thanks! It works!

**Comment by renrutsirhc** [comment]
This is an old post now, but in case anyone else comes accross this, there is a much simpler way to do this using the FormatString property on the NumericUpDown:

FormatString="0 mm"
FormatString="0.0 cm"
FormatString="0.00 in"
etc.

**Comment by stevemonaco** [comment]
In 11.1, `NumericUpDown` has `InnerLeftContent` and `InnerRightContent` on the control itself. Being able to text select the units (as you can do with `FormatString`) can be confusing to the user.
